src/shogun/features/LBPPyrDotFeatures.cpp

```diff
@@ -166,10 +166,13 @@ uint32_t* CLBPPyrDotFeatures::get_image(int32_t index, int32_t& width, int32_t&
 	return img;
 }
 
-float64_t CLBPPyrDotFeatures::dense_dot(int32_t vec_idx1, const float64_t* vec2, int32_t vec2_len) const
+float64_t
+CLBPPyrDotFeatures::dot(int32_t vec_idx1, const SGVector<float64_t> vec2) const
 {
-	if (vec2_len != vec_nDim)
-		SG_ERROR("Dimensions don't match, vec2_dim=%d, vec_nDim=%d\n", vec2_len, vec_nDim)
+	REQUIRE(
```

Comment:
could we maybe pass this type of check/warning on to `linalg`?

Reply:
yea, it's possible at some places